javascript/ql/lib/semmle/javascript/security/performance/ReDoSUtil.qll

```diff
@@ -537,7 +579,14 @@ private State after(RegExpTerm t) {
 predicate delta(State q1, EdgeLabel lbl, State q2) {
   exists(RegexpCharacterConstant s, int i |
     q1 = Match(s, i) and
-    lbl = Char(s.getValue().charAt(i)) and
+    (
+      not RegExpFlags::isIgnoreCase(s.getRootTerm()) and
+      lbl = Char(s.getValue().charAt(i))
+      or
+      // normalizing to lower case if ignorecase flag is set
+      RegExpFlags::isIgnoreCase(s.getRootTerm()) and
+      exists(string c | c = s.getValue().charAt(i) | lbl = Char(c.toLowerCase()))
+    ) and
```

Comment:
Should these labels also include flags?

Reply:
That shouldn't be needed. 
When we are in an "ignore-case" regexp i make sure to always create the lower-case of these labels (see the IPA constructor for `Char`).  
There should be no difference between a lowercase `Char`, and a `Char` that has been normalized to lowercase. 

That was I got as few labels as possible, which helps performance.